snapcraft/linters/base.py

```diff
@@ -72,12 +73,11 @@ def __str__(self):
 
         return msg
 
-    class Config:
-        """Pydantic model configuration."""
-
-        validate_assignment = True
-        extra = "forbid"
-        alias_generator = lambda s: s.replace("_", "-")  # noqa: E731
+    model_config = ConfigDict(
+        validate_assignment=True,
+        extra="forbid",
+        alias_generator=lambda s: s.replace("_", "-"),
```

Comment:
```suggestion
        alias_generator=base.alias_generator,
```

(with a higher up `from craft_application.models import base`)